import-export-cli/integration/testutils/throttlePolicy_testUtils.go

```diff
@@ -0,0 +1,244 @@
+	//case apim.CustomThrottlePolicyType:
+	//	exportedPolicy.Subtype = customPolicySubType
+	//case apim.ApplicationThrottlePolicyType:
+	//	exportedPolicy.Subtype = applicationPolicySubType
+	//}
+
+	exportedPolicy.Version = "v4.1.0"
+
+	policyData := client.GenerateSampleThrottlePolicyData(policyType)
+	policyMap, _ := ThrottlePolicyStructToMap(policyData)
+	var yamlMap yaml.MapSlice
+	yamlBytes, err := yaml.Marshal(policyMap)
+	err = yaml.Unmarshal(yamlBytes, &yamlMap)
+	if err != nil {
+		return "", err
+	}
+	exportedPolicy.Data = yamlMap
+	policyMarshaledData, _ := yaml.Marshal(exportedPolicy)
+
+	tmpDir, err := ioutil.TempDir("", "apim")
+	if err != nil {
+		_ = os.RemoveAll(tmpDir)
+		return "", err
+	}
+	tempFilePath := tmpDir + "/temp.yaml"
+	err = ioutil.WriteFile(tempFilePath, policyMarshaledData, os.ModePerm)
+	if err != nil {
+	}
+	return tempFilePath, err
+}
+
+// Adds a new Throttling Policy to an env
+func AddNewThrottlePolicy(t *testing.T, client *apim.Client, username, password, policyType string) interface{} {
+	client.Login(username, password)
+	generatedPolicy := client.GenerateSampleThrottlePolicyData(policyType)
+	addedPolicy := client.AddThrottlePolicy(t, generatedPolicy, policyType)
+	return addedPolicy
+}
+
+// Exports Throttling Policy from an env
+func exportThrottlePolicy(t *testing.T, name, env string) (string, error) {
+	output, err := base.Execute(t, "export", "policy", "rate-limiting", "-n", name, "-e", env, "-k", "--verbose")
+	return output, err
+}
+
+// Imports Throttling policy to an env
+func importThrottlePolicy(t *testing.T, args *ThrottlePolicyImportExportTestArgs) (string, error) {
+	var output string
+	var err error
+	if args.Update {
+		output, err = base.Execute(t, "import", "policy", "rate-limiting", "-e", args.DestAPIM.GetEnvName(), "-f", args.ImportFilePath, "-u")
+	} else {
+		output, err = base.Execute(t, "import", "policy", "rate-limiting", "-e", args.DestAPIM.GetEnvName(), "-f", args.ImportFilePath)
+	}
+	return output, err
+}
+
+// Retrieve Throttling policy by name
+func getThrottlingPolicyByName(t *testing.T, args *ThrottlePolicyImportExportTestArgs, throttlePolicyName, policyType string) (map[string]interface{}, error) {
+	client := args.DestAPIM
+	uuid := client.GetThrottlePolicyID(t, args.Admin.Username, args.Admin.Password, throttlePolicyName, policyType)
+	policy := client.GetThrottlePolicy(uuid, policyType)
+	client.DeleteThrottlePolicy(uuid, policyType)
+	return ThrottlePolicyStructToMap(policy)
+}
+
+// Validates whether throttling policies are equal
+func ValidatePoliciesEqual(t *testing.T, args *ThrottlePolicyImportExportTestArgs, importedPolicy map[string]interface{}) {
+	exportedPolicy := args.Policy
+	exportedPolicy[policyIDKey] = policyIDReset
+	importedPolicy[policyIDKey] = policyIDReset
+	assert.Equal(t, exportedPolicy, importedPolicy)
+}
+
+// Converts Throttling policy struct to map
+func ThrottlePolicyStructToMap(policy interface{}) (map[string]interface{}, error) {
```

Comment:
I think these utils functions are reusable for any kind of policy since we are having a type` map[string]interface{}`. What do you think @wasuradananjith ?

Reply:
Made reusable with 73381a57ed995c647ea2aa52394a7982f008363c